File: RESTful API Gin Gonic/src/repositories/employeeRepository.go
package repositories

import (
	"gin-gonic/src/entities"

	"gorm.io/gorm"
)

type IEmployeeRepository interface {
	FindAllColumnsLimit(row int) []entities.Employee
	FindHalfColumnLimit(row int) []entities.EmployeeHalf
}

type EmployeeRepository struct {
	db *gorm.DB
}

func NewEmployeeRepository(db *gorm.DB) IEmployeeRepository {
	return &EmployeeRepository{db}
}

func (repository *EmployeeRepository) FindAllColumnsLimit(row int) []entities.Employee {	
	var employees []entities.Employee
	repository.db.Raw("SELECT * FROM employee ORDER BY employee_id ASC LIMIT ?", row).Scan(&employees)
	return employees
}

func (repository *EmployeeRepository) FindHalfColumnLimit(row int) []entities.EmployeeHalf {
	var employees []entities.EmployeeHalf
	repository.db.Raw("SELECT employee_id, first_name, last_name, email, gender, date_of_birth FROM employee ORDER BY employee_id ASC LIMIT ?", row).Scan(&employees)
	return employees
}

File: RESTful API Spring Boot/src/main/java/com/springboot/restapi/repositories/EmployeeRepository.java
package com.springboot.restapi.repositories;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import com.springboot.restapi.entities.Employee;

@Repository
public interface EmployeeRepository extends JpaRepository<Employee, Long> {
    @Query(value="SELECT e.* FROM Employee e ORDER BY e.employee_id ASC LIMIT ?1", nativeQuery=true)
    List<Employee> findAllColumnsLimit(Integer limit);
}